### PR TITLE
Adds jsonPath to error messages, so errors can be pinpointed.

### PR DESCRIPTION
**Pull request by magnarn**
The jsonPath can be used to provide better error messages to users trying
to create valid jsonLogic expressions

Also:
Fixes some exception messages to use the right operator. ("none", "or").
Aligns behavior to reference implementation for sums of unparseable strings,
as well as for between comparisons of > and >=, and for comparison operators
with arrays of more than three values.
Handles edge cases of + and * the way jsonlogic.com does

### CODE REVIEW

File: src/test/java/io/github/jamsesso/jsonlogic/ParseErrorTests.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      fail("Expected an exception at " + jsonPath);
+    } catch (JsonLogicException e) {
+      assertEquals(e.getMessage(), jsonPath, e.getJsonPath());
+      assertEquals(msg, e.getMessage());
+    }
+  }
+
+  @Test
+  public void testJsonPath() {
+    testPath("{}", "$", "objects must have exactly 1 key defined, found 0");
```
Comment (jamsesso) [inline src/test/java/io/github/jamsesso/jsonlogic/ParseErrorTests.java:34]: I know this will be a huge pain in the ass, but could you put these into a fixtures file and then iterate over the file printing any failures that happen?

```
[
  {
    "test": "{}",
    "data": null,
    "expectedPath": "$",
    "expectedError": "obects must have exactly 1 key defined, found 0"
  },
  // etc...
]
```

followed by updating the test (pseudo-code):

```
for each fixture in parseJson(readFile("jsonpath-fixtures.json")) {
  result = testPath(fixture.test, fixture.expectedPath, fixture.expectedError, fixture.data);
  if (!result) {
    failures.add(fixture);
  }
}

if (failures.length() > 0) {
  // print all failures.
}
```

There is a similar thing being done here: https://github.com/jamsesso/json-logic-java/blob/main/src/test/java/io/github/jamsesso/jsonlogic/FixtureTests.java

File: src/main/java/io/github/jamsesso/jsonlogic/evaluator/expressions/LogicExpression.java [hunk 2]
```diff
@@ -23,16 +23,17 @@ public String key() {
...
-      throw new JsonLogicEvaluationException("and operator expects at least 1 argument");
+      throw new JsonLogicEvaluationException(key() + " operator expects at least 1 argument", jsonPath);
     }
 
     Object result = null;
 
+    int index = 0;
     for (JsonLogicNode element : arguments) {
-      result = evaluator.evaluate(element, data);
+      result = evaluator.evaluate(element, data, String.format("%s[%d]", jsonPath, index));
```
[the remark this answers is not in the document]
Comment (magnarn) [reply src/main/java/io/github/jamsesso/jsonlogic/evaluator/expressions/LogicExpression.java:36]: Thanks. Added a test for this.

**Comment by jamsesso** [comment]
Can you [bump the version](https://github.com/jamsesso/json-logic-java/blob/main/build.gradle#L8) to `1.1.0-SNAPSHOT` for me? When merged, it will publish as 1.1.0.

**Review by jamsesso — APPROVED** [review]

**Comment by jamsesso** [comment]
1.1.0 was published to maven central. Thanks for the contribution :)

**Comment by magnarn** [comment]
Sorry to bother you, but did you see the bugfix PR?
https://github.com/jamsesso/json-logic-java/pull/55

**Comment by justinabrahms** [comment]
In the future, it would be awesome if the major version was updated instead to indicate the backwards incompatibility.
